lib/inject.rb

```diff
@@ -0,0 +1,22 @@
+
+
+
+class Array
+
+  def new_inject(number = nil)
+
+    arr = self.dup
+
+    if number == nil
+      result = arr[0]
+      arr.shift
```

Comment:
These two lines can be brought together - `result = arr.shift` as `arr.shift` is equal to `arr[0]`